In [1]:
# import necessary packages
import pandas as pd
from sqlalchemy import create_engine

In [ ]:
# install mysql client
!pip install mysqlclient


In [3]:
engine = create_engine('mysql+mysqldb://admin:isba_4715@isba-dev-01.c3w8kciauusq.us-east-1.rds.amazonaws.com/EventData')

Business Question: What venues in Los Angeles are hosting the most EDM events?

In [6]:
# set query 1
query_1 = """
WITH venue_event_counts AS (
    SELECT venue, COUNT(eventid) AS event_count,
           RANK() OVER (ORDER BY COUNT(eventid) DESC) AS venue_rank
    FROM Events
    WHERE venue <> 'Los Angeles'
    GROUP BY venue
)
SELECT venue, event_count, venue_rank
FROM venue_event_counts
ORDER BY venue_rank;
"""


In [7]:

# execute the query and save results to top_venues_df
top_venues_df = pd.read_sql(query_1, engine)

# display the df
print(top_venues_df)


                           venue  event_count  venue_rank
0                     Academy LA           24           1
1                Sound Nightclub           19           2
2                 Time Nightclub           15           3
3                           1720           12           4
4               Avalon Hollywood           10           5
..                           ...          ...         ...
96                      Echoplex            1          49
97   Dirty Laundry (Boiler Room)            1          49
98                  197 E 2nd St            1          49
99                Elevate Lounge            1          49
100             The Speakeasy OC            1          49

[101 rows x 3 columns]


Insight: The top 5 venues holding the most events are Academy LA, Sound Nightclub, Time Nightclub, 1720, and Avalon Hollywood.

Recommendation: We should look into these top venues and do more research on the genres of the events they are holding to determine which of them we should partner with to sell tickets.

Prediction: If we partner with some of these venues to sell tickets, we will generate more sales as they are holding the most events.

Business Question: What are the genres of the events that these top venues are holding? Which of them are holding mostly house music events as this is the genre we are targeting.

In [8]:
# set query 2
query_2 = """
WITH venue_event_counts AS (
    SELECT venue, COUNT(eventid) AS event_count,
           RANK() OVER (ORDER BY COUNT(eventid) DESC) AS venue_rank
    FROM Events
    WHERE venue <> 'Los Angeles'
    GROUP BY venue
),
top_venues AS (
    SELECT venue
    FROM venue_event_counts
    WHERE venue_rank <= 5
),
ranked_genres AS (
    SELECT tv.venue, g.genre,
           RANK() OVER (PARTITION BY tv.venue ORDER BY COUNT(eg.eventid) DESC) AS genre_rank,
           COUNT(eg.eventid) AS event_count
    FROM top_venues tv
    JOIN Events e ON tv.venue = e.venue
    JOIN EventGenres eg ON e.eventid = eg.eventid
    JOIN Genres g ON eg.genre_id = g.genre_id
    GROUP BY tv.venue, g.genre
)
SELECT venue, genre, event_count
FROM ranked_genres
WHERE genre_rank <= 3
ORDER BY venue, genre_rank;
"""

In [9]:
# execute the query and save results to top_venues_df
top_genres_df = pd.read_sql(query_2, engine)

# display the df
print(top_genres_df)


               venue                      genre  event_count
0               1720                    dubstep            3
1               1720             downtempo bass            2
2               1720                    psybass            2
3               1720             gaming dubstep            2
4               1720             riddim dubstep            2
5         Academy LA                 tech house            5
6         Academy LA              electro house            5
7         Academy LA                        edm            4
8         Academy LA                  pop dance            4
9   Avalon Hollywood          progressive house            2
10  Avalon Hollywood           uplifting trance            2
11  Avalon Hollywood                     trance            2
12  Avalon Hollywood                      house            2
13  Avalon Hollywood                     techno            2
14   Sound Nightclub              melodic house            6
15   Sound Nightclub    

Insight: The venues who are holding the most events under the house genre are Academy LA, Sound Nightclub, and Time Nightclub.

Recommendation: We should look to partner with these 3 venues since we are targeting house music.

Prediction: If we partner with these venues, we will generate sales and become a leading ticket site for Los Angeles house music events.